### PR TITLE
Trenches - Add settings to configure dig/remove trenches durations

### PR DESCRIPTION
**Pull request by Mysteryjuju**
**When merged this pull request will:**
- Allow to customize trenches digging/removing durations.

### CODE REVIEW

File: addons/trenches/stringtable.xml [hunk 1]
```diff
@@ -239,5 +239,41 @@
...
+            <English>Big trench dig duration</English>
+            <French>Durée d'excavation d'une grande tranchée</French>
+        </Key>
+        <Key ID="STR_ACE_Trenches_BigEnvelopeDigDuration_Description">
+            <English>Big trench dig duration in seconds.</English>
+            <French>Définit la durée nécessaire pour déployer une grande tranchées (en secondes).</French>
+        </Key>
+        <Key ID="STR_ACE_Trenches_BigEnvelopeRemoveDuration_DisplayName">
+            <English>Big trench remove duration</English>
+            <French>Durée de retrait d'une grande tranchée</French>
```
Comment (Elgin675) [inline addons/trenches/stringtable.xml:272]:
```suggestion
            <French>Grandes tranchées - durée de retrait</French>
```

File: addons/trenches/stringtable.xml [hunk 2]
```diff
@@ -239,5 +239,41 @@
...
+        </Key>
+        <Key ID="STR_ACE_Trenches_BigEnvelopeRemoveDuration_Description">
+            <English>Big trench remove duration in seconds.</English>
+            <French>Définit la durée nécessaire pour retirer une grande tranchées (en secondes).</French>
```
Comment (Elgin675) [inline addons/trenches/stringtable.xml:276]:
```suggestion
            <French>Définit le temps nécessaire pour le retrait des grandes tranchées (en secondes).</French>
```

File: addons/trenches/stringtable.xml [hunk 3]
```diff
@@ -239,5 +239,41 @@
...
+            <English>Small trench dig duration in seconds.</English>
+            <French>Définit la durée nécessaire pour déployer une petite tranchées (en secondes).</French>
+        </Key>
+        <Key ID="STR_ACE_Trenches_SmallEnvelopeRemoveDuration_DisplayName">
+            <English>Small trench remove duration</English>
+            <French>Durée de retrait d'une petite tranchée</French>
+        </Key>
+        <Key ID="STR_ACE_Trenches_SmallEnvelopeRemoveDuration_Description">
+            <English>Small trench remove duration in seconds.</English>
+            <French>Définit la durée nécessaire pour retirer une petite tranchées (en secondes).</French>
```
Comment (Elgin675) [inline addons/trenches/stringtable.xml:260]:
```suggestion
            <French>Définit le temps nécessaire pour le retrait des petites tranchées (en secondes).</French>
```

File: addons/trenches/stringtable.xml [hunk 4]
```diff
@@ -239,5 +239,41 @@
...
+        </Key>
+        <Key ID="STR_ACE_Trenches_BigEnvelopeDigDuration_DisplayName">
+            <English>Big trench dig duration</English>
+            <French>Durée d'excavation d'une grande tranchée</French>
```
Comment (Elgin675) [inline addons/trenches/stringtable.xml:264]:
```suggestion
            <French>Grandes tranchées - durée d'excavation</French>
```

File: addons/trenches/stringtable.xml [hunk 5]
```diff
@@ -239,5 +239,41 @@
...
+        </Key>
+        <Key ID="STR_ACE_Trenches_BigEnvelopeDigDuration_Description">
+            <English>Big trench dig duration in seconds.</English>
+            <French>Définit la durée nécessaire pour déployer une grande tranchées (en secondes).</French>
```
Comment (Elgin675) [inline addons/trenches/stringtable.xml:268]:
```suggestion
            <French>Définit le temps nécessaire au déploiement des grandes tranchées (en secondes).</French>
```

File: addons/trenches/initSettings.sqf [hunk 6]
```diff
@@ -0,0 +1,36 @@
+// Trenches dig/remove durations
+[
+    QGVAR(smallEnvelopeDigDuration), 
+    "SLIDER",
```
Comment (mharis001) [inline addons/trenches/initSettings.sqf:4]: Consider using the `TIME` setting type.

File: addons/trenches/stringtable.xml [hunk 7]
```diff
@@ -239,5 +239,41 @@
             <Chinese>移除掩體中</Chinese>
             <Turkish>Siper Kaldırılıyor</Turkish>
         </Key>
+        <Key ID="STR_ACE_Trenches_Category">
+            <English>ACE Trenches</English>
+            <French>ACE Tranchées</French>
+        </Key>
+        <Key ID="STR_ACE_Trenches_SmallEnvelopeDigDuration_DisplayName">
+            <English>Small trench dig duration</English>
```
Comment (mharis001) [inline addons/trenches/stringtable.xml:247]: Title case for setting display names.

File: addons/trenches/stringtable.xml [hunk 8]
```diff
@@ -239,5 +239,41 @@
...
+            <French>Durée d'excavation d'une petite tranchée</French>
+        </Key>
+        <Key ID="STR_ACE_Trenches_SmallEnvelopeDigDuration_Description">
+            <English>Small trench dig duration in seconds.</English>
```
Comment (mharis001) [inline addons/trenches/stringtable.xml:251]: Maybe: "Time, in seconds, required to dig a small trench."

File: addons/trenches/stringtable.xml [hunk 9]
```diff
@@ -239,5 +239,41 @@
...
+            <French>Définit la durée nécessaire pour déployer une petite tranchées (en secondes).</French>
+        </Key>
+        <Key ID="STR_ACE_Trenches_SmallEnvelopeRemoveDuration_DisplayName">
+            <English>Small trench remove duration</English>
```
Comment (mharis001) [inline addons/trenches/stringtable.xml:255]: Maybe: "Time, in seconds, required to remove a small trench."

File: addons/trenches/initSettings.sqf (hunk 6 above)
Comment (Mysteryjuju) [reply addons/trenches/initSettings.sqf:4]: Updated.

File: addons/trenches/stringtable.xml (hunk 7 above)
Comment (Mysteryjuju) [reply addons/trenches/stringtable.xml:247]: Updated to title case

File: addons/trenches/stringtable.xml (hunk 8 above)
Comment (Mysteryjuju) [reply addons/trenches/stringtable.xml:251]: Updated

File: addons/trenches/stringtable.xml (hunk 9 above)
Comment (Mysteryjuju) [reply addons/trenches/stringtable.xml:255]: I updated description with this string. And strings for big trench too.

File: addons/trenches/initSettings.sqf [hunk 10]
```diff
@@ -0,0 +1,36 @@
+// Trenches dig/remove durations
+[
+    QGVAR(smallEnvelopeDigDuration), 
+    "TIME", 
+    [LSTRING(SmallEnvelopeDigDuration_DisplayName), LSTRING(SmallEnvelopeDigDuration_Description)],
+    LSTRING(Category),
+    [5, 600, 20], 
+    true
+] call CBA_Settings_fnc_init;
```
Comment (mharis001) [inline addons/trenches/initSettings.sqf:9]: `CBA_fnc_addSetting` here and below.

Comment (Mysteryjuju) [reply addons/trenches/initSettings.sqf]: Updated

**Pull request merged**